crates/holochain_conductor_api/src/admin_interface.rs

```diff
@@ -538,51 +557,148 @@ pub enum AppStatusFilter {
+
+    /// The number of seconds that the token should be valid for. After this number of seconds, the
+    /// token will no longer be accepted by the conductor.
+    ///
+    /// This is 30s by default which is reasonably high but with [IssueAppAuthenticationTokenPayload::single_use]
+    /// set to `false`, the token will be invalidated after the first use anyway.
+    ///
+    /// Set this to 0 to create a token that does not expire.
+    // #[serde(default = "30")]
+    pub expiry_seconds: u64,
+
+    /// Whether the token should be single-use. This is `true` by default and will cause the token
+    /// to be invalidated after the first use.
+    ///
+    /// Set this to `false` to allow the token to be used multiple times.
+    // #[serde(default = "true")]
+    pub single_use: bool,
+}
+
+impl IssueAppAuthenticationTokenPayload {
+    /// Create a new payload for issuing a connection token for the specified app.
+    ///
+    /// The token will be valid for 30 seconds and for a single use.
+    pub fn for_installed_app_id(installed_app_id: InstalledAppId) -> Self {
+        installed_app_id.into()
+    }
+
+    /// Set the expiry time for the token.
+    pub fn expiry_seconds(mut self, expiry_seconds: u64) -> Self {
+        self.expiry_seconds = expiry_seconds;
+        self
+    }
+
+    /// Set whether the token should be single-use.
+    pub fn single_use(mut self, single_use: bool) -> Self {
+        self.single_use = single_use;
+        self
+    }
+}
+
+impl From<InstalledAppId> for IssueAppAuthenticationTokenPayload {
+    fn from(installed_app_id: InstalledAppId) -> Self {
+        // Defaults here should match the serde defaults in the struct definition
+        Self {
+            installed_app_id,
+            expiry_seconds: 30,
+            single_use: true,
+        }
+    }
+}
+
+/// A token issued by the conductor that can be used to authenticate a connection to an app interface.
+pub type AppAuthenticationToken = Vec<u8>;
+
+/// Response payload for [AdminResponse::AppAuthenticationTokenIssued].
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct AppAuthenticationTokenIssued {
+    /// A token issued by the conductor that can be used to authenticate a connection the an app interface.
+    /// This is expected to be passed from the caller of the admin interface to the client that will
+    /// use the app interface. It should be treated as secret and kept from other parties.
+    pub token: AppAuthenticationToken,
+
+    /// The timestamp after which Holochain will consider the token invalid. This should be
+    /// considered informational only and the client should not rely on the token being accepted
+    /// following a client-side check that the token has not yet expired.
+    ///
+    /// If the token was created with [IssueAppAuthenticationTokenPayload::expiry_seconds] set to `0`
+    /// then this field will be `None`.
+    // TODO Kitsune type used in the conductor interface
```

Comment:
Is the TODO to replace `kitsune_p2p::Timestamp`? If so, how about the SystemTime or a std Timestamp?

Reply:
I started with `SystemTime` but these types either don't serialize or don't serialize to something that is easy to work with in JavaScript. Which is why I think `kitsune_p2p::Timestamp` is used in so many places across our public interface. But really there's no good reason to use a supposedly 'Kitsune' specific type here. I'm not sure what the right thing to do is here, but I'm sure that this type has leaked because it is useful and it doesn't really belong here.

Reply:
Or it doesn't belong in Kitsune and should be in a common crate somewhere... One of several tangles that needs to be untangled if we want to separate out Kitsune from the monster-repo :)